File: app/src/main/java/de/bitb/astroskop/injection/modules/ApplicationModule.java
package de.bitb.astroskop.injection.modules;


import dagger.Module;
import dagger.Provides;
import de.bitb.astroskop.AstroApplication;
import de.bitb.astroskop.injection.ApplicationScope;

@Module
public class ApplicationModule {

    private final AstroApplication application;

    public ApplicationModule(AstroApplication application) {
        this.application = application;
    }

    @Provides
    @ApplicationScope
    AstroApplication provideApplication() {
        return application;
    }

}

File: app/src/main/java/de/bitb/astroskop/injection/modules/ApplicationModule.kt
package de.bitb.astroskop.injection.modules


import dagger.Module
import dagger.Provides
import de.bitb.astroskop.BaseApplication
import javax.inject.Singleton

@Module
class ApplicationModule(private val application: BaseApplication) {

    @Provides
    @Singleton
    internal fun provideApplication(): BaseApplication {
        return application
    }

}